JavaScript-Core-1-Ikaneng-Khoza-Coursework-Week3-London8

ikanengKhoza opened the pull request:
Completed the excercises however I could not solve one excersice in extra folder

askinekincihf commented on 1-exercises/A-array-find/exercise.js:
```diff
@@ -4,6 +4,11 @@
 // write your code here
+function findLongNameThatStartsWithA(names) {
+  return names.find((name) => {
+    return (name.length > 7 && name.charAt(0)==="A");
```
Hi Murat, well done, I am really impressed!!  `charAt(0)==="A"`  👏 👏

askinekincihf commented on 1-exercises/D-array-filter/exercise.js:
```diff
@@ -8,7 +8,9 @@
 
-var pairsByIndex; // Complete this statement
+var pairsByIndex = pairsByIndexRaw.filter((pairArray) => {
+  return (typeof pairArray === "object" && pairArray!==null && pairArray.length === 2);
```
it is exactly what we are supposed to do. 👏 👏 
`typeof pairArray === "object" && pairArray !== null && pairArray.length === 2`

askinekincihf commented on 3-extra/1-card-validator.js:
```diff
@@ -0,0 +1,63 @@
+  });
+}
+
+function checkCardValidity(cardNumber) {
```
You have done a really great job! You have broken down all requirements into small functions and then combined all in one function which deserves 👏 👏 👏 .  I would only advise that you could use small letter for the first letter of variables.  This is very minor thing that I noticed. `  const NumberArray ` might be replaced as  ` const numberArray `

ikanengKhoza committed:
JS 1: 3 Homework Submission

Completed the excercises however I could not solve one excersice in extra folder

DelroyGayle commented on 1-exercises/B-while-loop/exercise.js:
```diff
@@ -2,12 +2,22 @@
+           evenNumber.push(i);
+           i += 2
+
+        } return evenNumber.toString();
```
Hi Ika, I cannot see where you add commas to your output.

DelroyGayle commented on 1-exercises/D-do-while/exercise.js:
```diff
@@ -7,8 +7,18 @@
+    
+        num += 2;
+        i += 1
+    }while (i < n){
```
There is no need to place the return statement within braces

DelroyGayle commented on 2-mandatory/2-retrying-random-numbers.js:
```diff
@@ -9,7 +9,16 @@ function generateRandomNumber() {
+   
+    }while( num <= 50) {
+    
+    return num; }
```
Again, no need to add 'return' within braces

DelroyGayle commented on 2-mandatory/4-stocks.js:
```diff
@@ -34,7 +34,21 @@ const CLOSING_PRICES_LAST_5_DAYS_FOR_ALL_STOCKS = [
+        sum += CLOSING_PRICES_LAST_5_DAYS_FOR_ALL_STOCKS[i][j];
+      }
+      average.push(
+        (sum / CLOSING_PRICES_LAST_5_DAYS_FOR_ALL_STOCKS[i].length).toFixed(2) *
```
As an alternative multiply the number by 100, then Math.round() it; then divide by 100

That way, you don't have to convert a number to a string , then multiply by 1 to convert it to a number again

DelroyGayle commented on 2-mandatory/4-stocks.js:
```diff
@@ -48,8 +62,12 @@ function getAveragePrices(closingPricesForAllStocks) {
-}
+    let newPriceChange = [];
+  for (let i = 0; i < closingPricesForAllStocks.length; i++) {
+    newPriceChange.push((closingPricesForAllStocks[i][4] - closingPricesForAllStocks[i][0]).toFixed(2) * 1);
```
See my above comments regarding Math.round()

DelroyGayle commented on 3-extra/1-factorial.js:
```diff
@@ -9,7 +9,11 @@
 function factorial(input) {
-    // TODO
+    
+     for (let i = input - 1; i >= 1; i--) {
```
I guess you could stop at i > 1 
since there is no need to multiply by 1
:)

DelroyGayle reviewed (commented):
Great work, Ika. Your work is to admired and inspires me to keep going. Thank you.

s-shah-coder commented on 1-exercises/A-undefined/exercise.js:
```diff
@@ -10,27 +10,32 @@
-let arr = [1,2,3];
+// Example 4 
+// not given a value.
+let arr = [1,2,3,4]; // now index 3 has a value of 4
```
Hi Ikaneng, very good start!
